refkit: upstream component update

pohly opened the pull request:
Makes us up-to-date on all components - well, as of a few days ago, and with the exception of meta-clang, which cannot be merged at this time because it breaks beignet-native.

okartau commented:
test this please

okartau commented:
test this please

pohly commented:
@rojkov can you review this? You are familiar with some of the problems which had to be solved as part of this PR.

I ended up adding some test enhancements because I was working on that in the same branch.

rojkov commented on meta-refkit-core/bbappends/meta-ros/recipes-ros/ros-comm/roscpp_1.11.21.bbappend:
```diff
@@ -0,0 +1,5 @@
+
+# Can be removed once https://github.com/ros/ros_comm/pull/1105 is in
+# meta-ros.
+SRC_URI_append = " file://roscpp-add-missing-header-for-writev.patch;striplevel=3"
```
Shouldn't this patch be a part of meta-refkit-industrial rather than meta-refkit-core?

rojkov commented on meta-refkit/lib/oeqa/selftest/cases/refkit_poky.py:
```diff
@@ -62,6 +62,14 @@ def test(self):
+                # Workaround for yocto-compat-layer.py not evaluating
+                # LAYERDEPENDS_security += "${@bb.utils.contains("DISTRO_FEATURES", "x11", "gnome-layer xfce-layer", "", d)}"
+                # in meta-security: add the layer ourselves in those cases where we know that it is needed.
+                if refkit_layer in ('meta-refkit',):
```
A string comparison would be enough here for now.

pohly commented:
On Thu, 2017-07-20 at 06:26 +0000, Dmitry Rozhkov wrote:
> @rojkov commented on this pull request.
> 
> 
> ______________________________________________________________________
> 
> In
> meta-refkit-core/bbappends/meta-ros/recipes-ros/ros-comm/roscpp_1.11.21.bbappend:
> 
> > @@ -0,0 +1,5 @@
> +FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
> +
> +# Can be removed once https://github.com/ros/ros_comm/pull/1105 is in
> +# meta-ros.
> +SRC_URI_append = " file://roscpp-add-missing-header-for-writev.patch;striplevel=3"
> 
> Shouldn't this patch be a part of meta-refkit-industrial rather than
> meta-refkit-core?

meta-refkit-industrial depends on meta-refkit-core, so it could be in
both. To me, "meta-refkit-core/bbappends" is where we do the dirty
workarounds for various upstream problems. I hadn't even considered
doing it in meta-refkit-industrial.

However, now that you mention it, it probably makes more sense in
meta-refkit-industrial, for the sake of consistency with the other,
already existing .bbappends there.

Another reason for moving it is that then yocto-compatible-layer.py
would have found the missing conditional in the _append operation. It
didn't flag that because the test against meta-refkit-core is done
without meta-ros in bblayers.conf.

I'll enhance the test and move the .bbappend.

pohly commented:
On Thu, 2017-07-20 at 06:32 +0000, Dmitry Rozhkov wrote:
> @rojkov commented on this pull request.
> 
> 
> ______________________________________________________________________
> 
> In meta-refkit/lib/oeqa/selftest/cases/refkit_poky.py:
> 
> > @@ -62,6 +62,14 @@ def test(self):
>                  - proper declaration of dependencies (because 'yocto-compat-layer.py --dependency' adds those)
>                  - parse and dependencies ('bitbake -S none world' must work)
>                  """
> +                # Workaround for yocto-compat-layer.py not evaluating
> +                # LAYERDEPENDS_security += "${@bb.utils.contains("DISTRO_FEATURES", "x11", "gnome-layer xfce-layer", "", d)}"
> +                # in meta-security: add the layer ourselves in those cases where we know that it is needed.
> +                if refkit_layer in ('meta-refkit',):
> 
> A string comparison would be enough here for now.

Okay. I kind of expected to add more layers, but that then turned out to
be unnecessary.

pohly commented:
@rojkov please have another look, I think I addressed your comments.

I also enhanced the meta-refkit-core testing and fixed all issues found by the more strict test.

rojkov commented on meta-refkit-industrial/recipes-ros/ros-comm/roscpp_1.11.21.bbappend:
```diff
@@ -0,0 +1,5 @@
+
+# Can be removed once https://github.com/ros/ros_comm/pull/1105 is in
+# meta-ros.
+SRC_URI_append = " file://roscpp-add-missing-header-for-writev.patch;striplevel=3"
```
I remember I had to add `_df-refkit-industrial` suffix to make bbappends compatible with poky tests. Is it still needed?

pohly commented:
On Thu, 2017-07-20 at 11:41 +0000, Dmitry Rozhkov wrote:
> +SRC_URI_append = " file://roscpp-add-missing-header-for-writev.patch;striplevel=3"
> 
> I remember I had to add _df-refkit-industrial suffix to make bbappends
> compatible with poky tests. Is it still needed?

Yes, and the test just failed:

test_compat_meta_refkit_industrial
(refkit_poky.TestRefkitPokySignatures) ... FAIL (144.438s)

Is this all, i.e. can I merge once this issue is fixed?

rojkov commented:
Yes, apart from the suffix the PR looks good to me.

On Thu, Jul 20, 2017 at 2:46 PM, Patrick Ohly <notifications@github.com>
wrote:

> On Thu, 2017-07-20 at 11:41 +0000, Dmitry Rozhkov wrote:
> > +SRC_URI_append = " file://roscpp-add-missing-header-for-writev.patch;
> striplevel=3"
> >
> > I remember I had to add _df-refkit-industrial suffix to make bbappends
> > compatible with poky tests. Is it still needed?
>
> Yes, and the test just failed:
>
> test_compat_meta_refkit_industrial
> (refkit_poky.TestRefkitPokySignatures) ... FAIL (144.438s)
>
> Is this all, i.e. can I merge once this issue is fixed?
>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/intel/intel-iot-refkit/pull/246#issuecomment-316679242>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACCYQplvcw7NUINJGo9mZsTMudgrVOYks5sPz4pgaJpZM4OX_KT>
> .
>

pohly commented:
Force pushed a fix and also added another test enhancement (commit 2a02b73930e22).

rojkov commented:
Looks good to me
